### DISCUSSION TITLE
Sort crates by number of dependents

### DISCUSSION THREAD

**Discussion by amab8901** [rust-lang/crates.io#8741] [Feature Requests]
When I browse crates.io, I wanna discover new crates that are likely to have valuable use-cases in my project (such as itertools, tokio, serde, cargo-chef, smallvec, etc). The best way I know to achieve this right now, is to "Browse All Crates" and sort the results by "Recent Downloads" or "All-time Downloads", interpreting these as metrics of popularity. But I realized that popularity can be more accurately measured by the number of dependents.

# Example:
https://crates.io/crates/thiserror-impl
this crate has 200 million all-time downloads and 31 million recent downloads. By these metrics, one might think "wow this is heavily used by Rust developers, so I should probably have a look at it since it may come in handy in my own projects!". But when we look at the number of dependents (https://crates.io/crates/thiserror-impl/reverse_dependencies), there's only 6 of them. If we had sorted the crates by number of dependents, then I wouldn't get a crate in my search results that is unlikely to ever be added in my Cargo.toml file.

**Comment by Turbo87** [comment]
calculating the number of reverse dependencies is unfortunately one of our most expensive database queries, which currently makes it unfeasible to sort by these numbers... :-/

**Reply by amab8901** [reply]
> calculating the number of reverse dependencies is unfortunately one of our most expensive database queries, which currently makes it unfeasible to sort by these numbers... :-/

How about caching it? A crate's popularity ranking in terms of dependents count is probably not that different today vs 10 days ago. Even if a crate may climb a couple steps up or down the ranking within that time, it still serves the purpose of helping me find the most useful crates more quickly. 

Calculate the dependency count for each crate maybe once every 10 days, and put the count as a metadata number for each crate. Then sort the crates by that metadata number.
